flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorContext.java

```diff
@@ -259,9 +265,9 @@ public void runInCoordinatorThread(Runnable runnable) {
     @Override
     public void close() throws InterruptedException {
         closed = true;
-        notifier.close();
-        coordinatorExecutor.shutdown();
-        coordinatorExecutor.awaitTermination(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
+        // Close quietly so the closing sequence will be executed completely.
+        closeQuietly(notifier);
```

Comment:
Can we use `IOUtils.closeAll` here as well?

```
        IOUtils.closeAll(
                notifier,
                () ->
                        shutdownExecutorForcefully(
                                coordinatorExecutor, Duration.ofMillis(Long.MAX_VALUE)));
```

Reply:
By default we "suppress" all exceptions, because the default `suppressedException` is `Exception.class` and we suppress all exception that can be assigned to `Exception`.

Reply:
I see. However, here I feel it is slightly more readable to just shutdown the two executors explicitly. It seems quite simple and straight forward compared with constructing two `AutoClosables` and invoke `closeAll()`.